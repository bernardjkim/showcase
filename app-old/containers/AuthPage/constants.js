/*
 *
 * AuthPage constants
 *
 */

export const CLEAR_ERRORS = 'app/AuthPage/CLEAR_ERRORS';

export const CREATE_USER = 'app/AuthPage/CREATE_USER';
export const CREATE_USER_SUCCESS = 'app/AuthPage/CREATE_USER_SUCCESS';
export const CREATE_USER_ERROR = 'app/AuthPage/CREATE_USER_ERROR';

export const CREATE_TOKEN = 'app/App/CREATE_TOKEN';
export const CREATE_TOKEN_SUCCESS = 'app/App/CREATE_TOKEN_SUCCESS';
export const CREATE_TOKEN_ERROR = 'app/App/CREATE_TOKEN_ERROR';
