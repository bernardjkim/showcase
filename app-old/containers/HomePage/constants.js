/*
 *
 * HomePage constants
 *
 */

export const CLEAR_STATE = 'app/HomePage/CLEAR_STATE';

export const LOAD_ARTICLES_ALL = 'app/HomePage/LOAD_ARTICLES_ALL';
export const LOAD_ARTICLES_ALL_ERROR = 'app/HomePage/LOAD_ARTICLES_ALL_ERROR';
export const LOAD_ARTICLES_ALL_SUCCESS =
  'app/HomePage/LOAD_ARTICLES_ALL_SUCCESS';

export const LOAD_NEXT = 'app/HomePage/LOAD_NEXT';
export const LOAD_NEXT_ERROR = 'app/HomePage/LOAD_NEXT_ERROR';
export const LOAD_NEXT_SUCCESS = 'app/HomePage/LOAD_NEXT_SUCCESS';
