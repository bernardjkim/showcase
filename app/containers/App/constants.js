/*
 * AppConstants
 * Each action has a corresponding type, which the reducer knows and picks up on.
 * To avoid weird typos between the reducer and the actions, we save them as
 * constants here. We prefix them with 'yourproject/YourComponent' so we avoid
 * reducers accidentally picking up actions they shouldn't.
 *
 * Follow this format:
 * export const YOUR_ACTION_CONSTANT = 'yourproject/YourContainer/YOUR_ACTION_CONSTANT';
 */

export const CREATE_TOKEN = 'app/App/CREATE_TOKEN';
export const CREATE_TOKEN_SUCCESS = 'app/App/CREATE_TOKEN_SUCCESS';
export const CREATE_TOKEN_ERROR = 'app/App/CREATE_TOKEN_ERROR';
