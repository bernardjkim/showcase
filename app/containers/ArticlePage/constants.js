/*
 *
 * ArticlePage constants
 *
 */

export const CREATE_COMMENT = 'app/ArticlePage/CREATE_COMMENT';
export const CREATE_COMMENT_SUCCESS = 'app/ArticlePage/CREATE_COMMENT_SUCCESS';
export const CREATE_COMMENT_ERROR = 'app/ArticlePage/CREATE_COMMENT_ERROR';

export const LIKE_ARTICLE = 'app/ArticlePage/LIKE_ARTICLE';
export const LIKE_ARTICLE_SUCCESS = 'app/ArticlePage/LIKE_ARTICLE_SUCCESS';
export const LIKE_ARTICLE_ERROR = 'app/ArticlePage/LIKE_ARTICLE_ERROR';

export const LOAD_ARTICLE = 'app/ArticlePage/LOAD_ARTICLE';
export const LOAD_ARTICLE_SUCCESS = 'app/ArticlePage/LOAD_ARTICLE_SUCCESS';
export const LOAD_ARTICLE_ERROR = 'app/ArticlePage/LOAD_ARTICLE_ERROR';

export const LOAD_COMMENTS = 'app/ArticlePage/LOAD_COMMENTS';
export const LOAD_COMMENTS_SUCCESS = 'app/ArticlePage/LOAD_COMMENTS_SUCCESS';
export const LOAD_COMMENTS_ERROR = 'app/ArticlePage/LOAD_COMMENTS_ERROR';
