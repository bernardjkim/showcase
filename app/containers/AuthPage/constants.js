/*
 *
 * AuthPage constants
 *
 */

export const CREATE_USER = 'app/AuthPage/CREATE_USER';
export const CREATE_USER_SUCCESS = 'app/AuthPage/CREATE_USER_SUCCESS';
export const CREATE_USER_ERROR = 'app/AuthPage/CREATE_USER_ERROR';
