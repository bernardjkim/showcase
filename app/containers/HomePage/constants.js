/*
 *
 * HomePage constants
 *
 */

export const LIKE_ARTICLE = 'app/HomePage/LIKE_ARTICLE';
export const LIKE_ARTICLE_SUCCESS = 'app/HomePage/LIKE_ARTICLE_SUCCESS';
export const LIKE_ARTICLE_ERROR = 'app/HomePage/LIKE_ARTICLE_ERROR';

export const LOAD_ARTICLE = 'app/HomePage/LOAD_ARTICLE';
export const LOAD_ARTICLE_SUCCESS = 'app/HomePage/LOAD_ARTICLE_SUCCESS';
export const LOAD_ARTICLE_ERROR = 'app/HomePage/LOAD_ARTICLE_ERROR';
