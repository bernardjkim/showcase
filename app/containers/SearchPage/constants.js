/*
 *
 * SearchPage constants
 *
 */

export const LOAD_ARTICLES = 'app/SearchPage/LOAD_ARTICLES';
export const LOAD_ARTICLES_SUCCESS = 'app/SearchPage/LOAD_ARTICLES_SUCCESS';
export const LOAD_ARTICLES_ERROR = 'app/SearchPage/LOAD_ARTICLES_ERROR';
