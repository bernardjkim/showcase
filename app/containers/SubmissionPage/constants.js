/*
 *
 * SubmissionPage constants
 *
 */

export const SUBMIT_FORM = 'app/SubmissionPage/SUBMIT_FORM';
export const SUBMIT_FORM_SUCCESS = 'app/SubmissionPage/SUBMIT_FORM_SUCCESS';
export const SUBMIT_FORM_ERROR = 'app/SubbmissionPage/SUBMIT_FORM_ERROR';
