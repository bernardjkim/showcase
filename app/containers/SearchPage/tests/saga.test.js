/**
 * Test sagas
 */

/* eslint-disable redux-saga/yield-effects */
import { takeLatest, all, put } from 'redux-saga/effects';
import { cloneableGenerator } from 'redux-saga/utils';
import searchPageSaga, { loadArticles, loadNext } from '../saga';
import { LOAD_ARTICLES, LOAD_NEXT } from '../constants';
import {
  loadArticlesSuccess,
  loadArticlesError,
  loadNextError,
  loadNextSuccess,
} from '../actions';

describe('loadArticles Saga', () => {
  const gen = cloneableGenerator(loadArticles)();
  const search = 'search';
  const offset = '0';

  const selectDescriptor = gen.next().value;
  expect(selectDescriptor).toMatchSnapshot();

  gen.next(search);
  gen.next(offset);
  const callDescriptor = gen.value;
  expect(callDescriptor).toMatchSnapshot();

  it('should dispatch the loadArticlesSuccess action if successful', () => {
    const searchResults = { hits: { hits: [] } };
    const articles = [];

    const putDescriptor = gen.clone().next(searchResults).value;
    expect(putDescriptor).toEqual(put(loadArticlesSuccess(articles)));
  });

  it('should dispatch the loadArticlesError action if error', () => {
    const error = new Error('Test Error');
    const putDescriptor = gen.clone().throw(error).value;
    expect(putDescriptor).toEqual(put(loadArticlesError(error)));
  });
});

describe('loadNext Saga', () => {
  const gen = cloneableGenerator(loadNext)();
  const search = 'search';
  const offset = '0';

  const selectDescriptor = gen.next().value;
  expect(selectDescriptor).toMatchSnapshot();

  gen.next(search);
  gen.next(offset);
  const callDescriptor = gen.value;
  expect(callDescriptor).toMatchSnapshot();

  it('should dispatch the loadNextSuccess action if successful', () => {
    const searchResults = { hits: { hits: [] } };
    const articles = [];

    const putDescriptor = gen.clone().next(searchResults).value;
    expect(putDescriptor).toEqual(put(loadNextSuccess(articles)));
  });

  it('should dispatch the loadNextError action if error', () => {
    const error = new Error('Test Error');
    const putDescriptor = gen.clone().throw(error).value;
    expect(putDescriptor).toEqual(put(loadNextError(error)));
  });
});

describe('searchPageSaga Saga', () => {
  const saga = searchPageSaga();

  it('should start task to watch for api request actions', () => {
    const takeLatestDescriptor = saga.next().value;
    expect(takeLatestDescriptor).toEqual(
      all([
        takeLatest(LOAD_ARTICLES, loadArticles),
        takeLatest(LOAD_NEXT, loadNext),
      ]),
    );
  });
});
