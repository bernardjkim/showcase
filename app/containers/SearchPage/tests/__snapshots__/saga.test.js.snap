// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "http://localhost:8080/api/article/search?q=q",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
